sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/SortBasedAggregate.scala

```diff
@@ -63,7 +68,13 @@ case class SortBasedAggregate(
   }
 
   protected override def doExecute(): RDD[InternalRow] = attachTree(this, "execute") {
-    child.execute().mapPartitions { iter =>
+    val numInputRows = longMetric("numInputRows")
+    val numOutputRows = longMetric("numOutputRows")
+    child.execute().mapPartitions { _iter =>
+      val iter = _iter.map { row =>
```

Comment:
it'd be great if we can avoid the extra iterator map, because an iterator wrapper actually introduces a lot of overhead.